doc/publishing-application-examples.md

```diff
@@ -0,0 +1,194 @@
+# Publishing application examples
+
+This documentation aims to explain integration steps for a publishing application to manage content items and links within the Publishing API. Included are request flow concepts and trivial code examples to aid the developer with integration steps.
+
+The publishing API provides a set of endpoints aimed at providing client publishing applications a way to draft and modify content and links and publish them.
+Content and links for a shared content item can be modified separately on different endpoints, these distinct changes will ultimately modify the downstream content item in the content store.
+
+## Authentication and audit trail
+
+Publishing applications may need to track the changes made to a content item against an authenticated user. In order to do this [integration with GDS-SSO is required](https://github.com/alphagov/gds-sso#integration-with-a-rails-3-app), this usually means implementing a persisted user model in the publishing application and applying a filter in the application controller to authenticate the user.
+The `User#uid` value can then be sent as a request parameter when drafting content to restrict access. eg.
+
+```
+  access_limited : ["8f3173bb-ee2a-48a2-aa86-f2f030830888"]
+```
+
+## Draft and Publish endpoint flow
+
+The Publishing API prescribes a draft-to-live workflow where the publishing application submits new draft content, this content may then be updated via the same endpoint and finally the content item can be published in a discreet endpoint request.
+
+```
+  Publishing Application                                 Publishing API
+
+  [ Initial draft ] -----------------------------------> [ PUT /v2/content/:content_id ]
+                           Responds with version                       |
```

Comment:
Does it respond with a version number only or with a full draft state that also contains a new version number?